src/content/reference/react/use.md

```diff
@@ -0,0 +1,769 @@
+
+function Message({ messagePromise }) {
+  const message = use(messagePromise);
+  const theme = use(ThemeContext);
+  //...
+```
+
+Unlike all other React Hooks, the `use` Hook can be called within conditional statements like loops and `if` statements. Like other React Hooks, the parent function of a Hook must be a React component or Hook.
+
+A component that passes a promise to `use` integrates with [`Suspense`](/reference/react/Suspense) and [`ErrorBoundary`](/reference/react/Component#catching-rendering-errors-with-an-error-boundary). When a pending promise is passed to `use`, the fallback of the closest Suspense Boundary will be displayed. When the promise is resolved, the component calling `use` will be rendered using the data from the resolved promise. The rendered component will then replace the Suspense Boundary fallback. If the promise passed to `use` is rejected, the fallback of the nearest Error Boundary will be displayed. 
+
+[See more examples below.](#usage)
+
+#### Parameters {/*parameters*/}
+
+* `resource`: this is the source of the data you want to read a value from. A resources can be a [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) or a [context](/learn/passing-data-deeply-with-context).
+
+#### Returns {/*returns*/}
+
+The `use` Hook returns the value that was read from the resource like the resolved value of a [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) or [context](/learn/passing-data-deeply-with-context).
+
+#### Caveats {/*caveats*/}
+
+* The parent function of the `use` Hook must be a React component or your own Hooks.
+* `use` is not recommended for data fetching from a [Server Component](/reference/react/components#server-components). `async` and `await` are preferred when fetching data from a Server Component.
+* When passing promises to `use`, you must store the promise in a state variable or context to avoid recreating the promise on every rerender. When invoking a function that returns a promise, you can to store the returned promise with a cache mechanism like React's `cache` function.
+
+---
+
+## Usage {/*usage*/}
+
+### Reading promises with `use` {/*reading-promises-with-use*/}
+
+`use` functions similarly to the `await` with JavaScript promises but is compatible with React's renderer and [`Suspense`](/reference/react/Suspense). Browser APIs or data fetching libraries that return a promise can be passed to `use` to read the value of the promise.
+
+In the example below we'll use <CodeStep step={1}>[`navigator.storage.estimate()`](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager/estimate)</CodeStep>, a browser API that provides information on how much storage is being used by the current page and how much space is available. <CodeStep step={1}>`navigator.storage.estimate()`</CodeStep> is an async function that returns a Promise which resolves once the information is available.
```

Comment:
Inconsistent capitalization of promise. I'm not sure which it should be

Reply:
Looks like another page [capitalizes Promise](https://react.dev/reference/react-dom/server/renderToReadableStream#returns) I've updated all the references to Promise on this page to be capitalized.

Seems like async and asynchronous are both used on various react.dev pages. asynchronous does seem more appropriate. I've updated all references to asynchronous instead of async unless otherwise required (e.g. the `async` keyword before a function).